src/gromo/growing_module.py

```diff
@@ -1183,32 +1183,78 @@ def reset_computation(self) -> None:
+            if True delete the extended_output_layer of this layer,
+            warning: this does not delete the extended_input_layer of the next layer
         """
         self.optimal_delta_layer = None
-        self.extended_input_layer = None
-        self.extended_output_layer = None
         self.scaling_factor = 0.0  # type: ignore
         # this type problem is due to the use of the setter to change the scaling factor
         self.parameter_update_decrease = None
         self.eigenvalues_extension = None
         self._pre_activity = None
         self._input = None
 
-        if include_previous and self.previous_module is not None:
-            if isinstance(self.previous_module, GrowingModule):
-                self.previous_module.extended_output_layer = None
-            elif isinstance(self.previous_module, AdditionGrowingModule):
-                raise NotImplementedError  # TODO
-            else:
-                raise NotImplementedError
+        # delete extended_output_layer
+        if include_output:
+            self.extended_output_layer = None
+
+        # delete previous module extended_output_layer
+        if self.extended_input_layer is not None:
+            # delete extended_input_layer
+            self.extended_input_layer = None
+            if self.previous_module is not None:
+                # normal behavior
+                if include_previous:
+                    if isinstance(self.previous_module, GrowingModule):
+                        self.previous_module.extended_output_layer = None
+                    elif isinstance(self.previous_module, AdditionGrowingModule):
+                        raise NotImplementedError  # TODO
+                    else:
+                        raise TypeError(
+                            f"Unexpected type for previous_module of {self.name}"
+                            f"got {type(self.previous_module)} instead of GrowingModule or AdditionGrowingModule."
+                        )
+                # risky behavior
+                else:  # include_previous is False
+                    if isinstance(self.previous_module, GrowingModule):
+                        if self.previous_module.extended_output_layer is not None:
+                            warnings.warn(
+                                f"The extended_input_layer of {self.name} has been deleted."
+                                f"However, the extended_output_layer associated stored in the previous module"
+                                f"named {self.previous_module.name} has not been deleted."
+                                "This may lead to errors when using extended_forward.",
+                                UserWarning,
+                            )
+                        # otherwise it is ok as user already deleted the extended_output_layer
+                    elif isinstance(self.previous_module, AdditionGrowingModule):
+                        raise NotImplementedError
```

Comment:
There is no reason for this elif statement. It will break the code because all GrowingModules are followed by AdditionGrowingModules in the graph implementation.

Reply:
Normally this `elif` statement won't break tour code as it was already there before the change of this PR.
The only thing that could break your code is that by default we do not delete ` self.extended_output_layer = None`.
The other question is what should we about other linked extension do when we call delete update on a module preceded by an `AdditionGroMo` ? I see to options:
(0. Do not decide yet an keep the  `raise`)
1. Do nothing (ie replace `raise NotImplementedError` by `return` or a warning) and let the user fully in charge of deleting the associated extensions.
2. Delete associated extension ie all previous extended output, all parallel extended input and maybe more as we could have skip connections...

For me the options 2 is more logical if there is an relatively easy way to implement it. Otherwise we could choose options 1. What do you think ?

Reply:
This elif was not there before. The same code block existed only inside the if statement of `include_previous` (line 1223) and it was not code breaking because I was not using the include_previous option. Now that it is checked everywhere any GrowingModule that has an AdditionGrowingModule as previous will break the code.

Reply:
I was always a fan of letting the user delete all previous updates on their own. Having to check backward is getting overly complicated. Personally, when I am using the code I find it easier to delete all extensions on each module individually to better keep track of what had been deleted and make sure that all updates are deleted (extension, optimal, etc.). Hopefully, we have set enough checks to inform the user if he has forgotten to delete something, with the warnings and errors in the forward function.

Reply:
In this case it make a lot of sense to choose to choose options 1. when `include_previous` is `False`. I will change it !